modules/dnn/src/op_halide.cpp

```diff
@@ -16,6 +16,8 @@
 #include <HalideRuntimeOpenCL.h>
 #endif  // HAVE_HALIDE
 
+#include <thread>
```

Comment:
> `<thread>`

This header should not be used unconditionally.
Otherwise this patch breaks some build modes like `OPENCV_DISABLE_THREAD_SUPPORT`

Put header under the same pre-processor guards similar to related code:

```
#ifdef HAVE_HALIDE
```

(move 2 lines upper)